tests/System.Buffers.Experimental.Tests/MemoryTests.cs

```diff
@@ -90,5 +90,161 @@ static void TestLifetime(OwnedBuffer<byte> owned)
                 var span = copyStoredForLater.Span;
             });
         }
+
+        [Fact]
+        public unsafe void IsSliceOf_PointerVersusArraySpansNeverMatch()
+        {
+            ulong[] data = new ulong[8];
+            fixed (ulong* ptr = data)
+            {
+                Span<ulong> arrayBased = data;
+                Span<ulong> pointerBased = new Span<ulong>(ptr, 8);
+
+                // comparing array to pointer says "no"
+                Assert.False(arrayBased.IsSliceOf(pointerBased));
```

Comment:
I guess that's a philosophical semantic question; in the original code where the `IsSliceOf` was adjacent to the `Slice` method, it had more knowledge of the internals - and we could say with certainty whether something could have come from a `Slice(...)` on another span. A Span into an array would never be confused with the Span of a pointer.

However, it is also valid to say that they're *equivalent*. At the time, this choice was partly to do with the fact that `Equals` leaked this information about whether they were the same implementation or not (I even filed a bug, IIRC). However, it looks like this has now been fixed - with `Equals` reporting NSE, and `==` working in the way you describe, with them comparing as equivalent even if array vs pointer.

So: I think you're right and this should be changed. At the time it was consistent with how `Span<T>` behaved, but `Span<T>` has *been changed* to not mis-behave like this. Will fix.